mlflow/tracing/fluent.py

```diff
@@ -792,3 +794,109 @@ def _merge_trace(
             **trace.info.request_metadata,
             **parent_trace.info.request_metadata,
         }
+
+
+def set_span_chat_messages(
+    span: LiveSpan,
+    messages: list[RequestMessage],
+):
+    """
+    Set the `mlflow.chat.messages` attribute on the specified span. This
+    attribute is used in the UI, and also by downstream applications that
+    consume trace data, such as MLflow evaluate.
+
+    Args:
+        span: The LiveSpan to add the attribute to
+        messages: A list of standardized chat messages (refer to the
+                 `spec <../llms/tracing/tracing-schema.html#chat-completion-spans>`_
+                 for details)
+
+    Example:
+
+    .. code-block:: python
+        :test:
+
+        import mlflow
+
+
+        @mlflow.trace
+        def f():
+            messages = [{"role": "user", "content": "hello"}]
+            span = mlflow.get_current_active_span()
+            mlflow.set_span_chat_messages(span, messages)
+            return 0
+
+
+        f()
+    """
+    try:
+        for message in messages:
+            if IS_PYDANTIC_V2:
+                RequestMessage.model_validate(message)
+            else:
+                RequestMessage.parse_obj(message)
+    except pydantic.ValidationError as e:
+        raise MlflowException.invalid_parameter_value(
+            "Chat messages must be a list of valid messages."
+        ) from e
+    span.set_attribute(SpanAttributeKey.CHAT_MESSAGES, messages)
```

Comment:
Is there any method in pydantic we can extend to implement this error conversion?

Reply:
actually i think we can just raise the base pydantic validation error, i don't think there's a benefit to doing the conversion

Reply:
When this code is running in a server, the error type (= status code) is important.

Reply:
i think these functions should only be called in clients (since they operate on livespans), but added back the conversion as there should be no harm. also added a couple tests for validation